targets/common

```diff
@@ -45,6 +58,10 @@ elif [ -z "$TARGETNOINSTALL" -o -n "$RESTOREHOSTBIN" ]; then
     done
     # Source the prerequisites
     for t in $REQUIRES; do
+	# If it is already provided change it to the target what provides it.
+        if grep -q "^$t=.*\$" "${PROVIDESFILE:-/dev/null}"; then
```

Comment:
No need to grep before the awk.  Store in a temporary variable and then say `t="${provider:-"$t"}"`.